nodebuilder/share/module.go

```diff
@@ -105,79 +110,70 @@ func ConstructModule(tp node.Type, cfg *Config, options ...fx.Option) fx.Option
 		fx.Provide(func(shrexSub *shrexsub.PubSub) shrexsub.BroadcastFn {
 			return shrexSub.Broadcast
 		}),
+	)
+
+	lightAndFullComponents := fx.Options(
+		fx.Provide(getters.NewIPLDGetter),
+		fx.Provide(peers.NewManager),
 		fx.Provide(
-			func(ctx context.Context, h host.Host, network modp2p.Network) (*shrexsub.PubSub, error) {
-				return shrexsub.NewPubSub(
-					ctx,
-					h,
-					network.String(),
-				)
+			func(host host.Host, network modp2p.Network) (*shrexnd.Client, error) {
+				cfg.ShrExNDParams.WithNetworkID(network.String())
+				return shrexnd.NewClient(cfg.ShrExNDParams, host)
+			},
+		),
+		fx.Provide(
+			func(host host.Host, network modp2p.Network) (*shrexeds.Client, error) {
+				cfg.ShrExEDSParams.WithNetworkID(network.String())
+				return shrexeds.NewClient(cfg.ShrExEDSParams, host)
 			},
 		),
+		fx.Provide(fx.Annotate(
+			getters.NewShrexGetter,
+			fx.OnStart(func(ctx context.Context, getter *getters.ShrexGetter) error {
+				return getter.Start(ctx)
+			}),
+			fx.OnStop(func(ctx context.Context, getter *getters.ShrexGetter) error {
+				return getter.Stop(ctx)
+			}),
+		)),
 	)
 
 	switch tp {
+	case node.Bridge:
+		return fx.Module(
+			"share",
+			baseComponents,
+			bridgeAndFullComponents,
+			fxutil.ProvideAs(func(getter *getters.StoreGetter) share.Getter {
+				return getter
+			}),
+		)
+	case node.Full:
+		return fx.Module(
+			"share",
+			baseComponents,
+			lightAndFullComponents,
+			bridgeAndFullComponents,
```

Comment:
are there no duplicates between the two? This is also confusing 😭

Reply:
No duplicates. It is just grouping of common components.

Reply:
Ok fine

Reply:
I've regrouped lightAndFullComponents a bit and renamed it. Hope it feels better now 🙂